app/lib/url.rb

```diff
@@ -46,7 +46,11 @@ def self.reaction(reaction)
   #
   # @param tag [Tag] the tag to create the URL for
   def self.tag(tag, page = 1)
-    url(["/t/#{CGI.escape(tag.name)}", ("/page/#{page}" if page > 1)].join)
+    url([tag_path(tag), ("/page/#{page}" if page > 1)].join)
+  end
+
+  def self.tag_path(tag)
```

Comment:
I wonder if it may be better to finally override `to_param` in the tag model. I added the following locally:

```ruby
def to_param
  CGI.escape(name)
end
```

This gives me all the basic helpers one would expect:

```
app.edit_tag_path(@tag)
#=> "/t/beginners/edit"

app.tag_path(@tag)
#=> "/t/beginners"
```

We could then include `ActionView::Helpers::UrlHelper` in `URL` to make use of this. 

But we'd have to check to make sure this doesn't break any existing code relying on `to_param` returning an id, so it's probably a bad idea. Anyway, ignore this, it's just me thinking out "loud".